File: Invert.java
package main.java.com.iiitb.imageEffectApplication.effects;
import com.iiitb.imageEffectApplication.exception.IllegalParameterException;
import com.iiitb.imageEffectApplication.libraryInterfaces.InvertInterface;
import com.iiitb.imageEffectApplication.libraryInterfaces.Pixel;
import com.iiitb.imageEffectApplication.service.LoggingService;
import com.iiitb.imageEffectApplication.baseEffects.PhotoEffect;

public class Invert implements PhotoEffect{
    public Pixel[][] apply(Pixel[][] image, String fileName, LoggingService loggingService){
        return InvertInterface.applyInvert(image);
    }
}

File: Invert.cpp
#include<vector>
//#include "libraryInterfaces_InvertInterface.h"
#include "Invert.h"
#include "../Pixel.h"

using namespace std;

void Invert(vector<vector<Pixel>>& imageVector){
    for(int i = 0; i < imageVector.size(); i++){
        for(int j = 0; j < imageVector[i].size(); j++){
            imageVector[i][j].r = 255 - imageVector[i][j].r;
            imageVector[i][j].g = 255 - imageVector[i][j].g;
            imageVector[i][j].b = 255 - imageVector[i][j].b;
        }
    }
}

